src/components/Inputs/BaseInput/BaseInput.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { GeneralProps } from '../types';
+
+interface BaseInputSpecificProps {
+  type: 'text';
+}
+
+type BaseInputProps = BaseInputSpecificProps & GeneralProps;
+
+interface ColorsMap {
+  [key: string]: string;
+}
+
+const BaseInput = ({
```

Comment:
Should we consider the possibility of having icons both left and right on this component?

Reply:
Yes the idea was to add those properties progressively as needed, like in the password input PR